db/gdocBlockParsersRawToEnriched.ts

```diff
@@ -0,0 +1,726 @@
+        return createError({
+            message: "Value is a string, not an object with properties",
+        })
+
+    if (raw.value.length === 0)
+        return createError({
+            message: "Recirc must have at least one item",
+        })
+
+    const title = raw.value[0].title
+    if (!title)
+        return createError({
+            message: "Title property is missing or empty",
+        })
+
+    if (!raw.value[0].list)
+        return createError({
+            message: "Recirc must have at least one entry",
+        })
+
+    const items: (EnrichedRecircItem | ParseError[])[] = raw.value[0].list.map(
+        (item): EnrichedRecircItem | ParseError[] => {
+            if (typeof item?.article !== "string")
+                return [
+                    {
+                        message:
+                            "Item is missing article property or it is not a string value",
+                    },
+                ]
+            if (typeof item?.author !== "string")
+                return [
+                    {
+                        message:
+                            "Item is missing author property or it is not a string value",
+                    },
+                ]
+            if (typeof item?.url !== "string")
+                return [
+                    {
+                        message:
+                            "Item is missing url property or it is not a string value",
+                    },
+                ]
+
+            const article = htmlToSimpleTextBlock(item.article)
+            const author = htmlToSimpleTextBlock(item.author)
+
+            const errors = article.parseErrors.concat(author.parseErrors)
+
+            if (errors.length > 0) return errors
+
+            return {
+                url: item.url,
+                article: article.value,
+                author: author.value,
+            }
+        }
+    )
+
+    const [errors, enrichedItems] = partition(
+        items,
+        (item: EnrichedRecircItem | ParseError[]): item is ParseError[] =>
+            isArray(item)
+    )
+
+    const flattenedErrors = errors.flat()
+    const parsedTitle = htmlToSimpleTextBlock(title)
+
+    return {
+        type: "recirc",
+        title: parsedTitle.value,
+        items: enrichedItems,
+        parseErrors: [...flattenedErrors, ...parsedTitle.parseErrors],
+    }
+}
+
+const parseText = (raw: RawBlockText): EnrichedBlockText => {
+    const createError = (
+        error: ParseError,
+        value: Span[] = []
+    ): EnrichedBlockText => ({
+        type: "text",
+        value,
+        parseErrors: [error],
+    })
+
+    if (typeof raw.value !== "string")
+        return createError({
+            message: "Value is a not a string",
+        })
+
+    const value = htmlToSpans(raw.value)
+
+    return {
+        type: "text",
+        value,
+        parseErrors: [],
+    }
+}
+
+const parseHeader = (raw: RawBlockHeader): EnrichedBlockHeader => {
```

Comment:
"header" should probably be renamed as "heading". Header = block / page header, heading ~ title of section